### PR TITLE
Merge changes from dd-trace-java 0.39.0

### PR DESCRIPTION
**Pull request by tylerbenson**
https://github.com/DataDog/dd-trace-java/releases/tag/v0.39.0

Lots of merge conflicts.  I hope everything was resolved ok.

### CODE REVIEW

**Comment by tylerbenson** [comment]
I'm still working with my team to get the CLA thing figured out...

File: dd-java-agent/instrumentation/jax-rs-annotations-2/src/test/groovy/JaxRsFilterTest.groovy [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+import datadog.trace.api.DDTags
+import datadog.trace.instrumentation.api.Tags
+import io.dropwizard.testing.junit.ResourceTestRule
+import javax.ws.rs.client.Entity
+import javax.ws.rs.container.ContainerRequestContext
+import javax.ws.rs.container.ContainerRequestFilter
+import javax.ws.rs.container.PreMatching
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+import javax.ws.rs.ext.Provider
```
Comment (trask) [inline dd-java-agent/instrumentation/jax-rs-annotations-2/src/test/groovy/JaxRsFilterTest.groovy]: It looks like import order was changed during merge conflict resolution. Do you have correct import layout for groovy (https://github.com/open-telemetry/opentelemetry-auto-instr-java/blob/master/CONTRIBUTING.md#intellij-idea)?

Comment (tylerbenson) [reply dd-java-agent/instrumentation/jax-rs-annotations-2/src/test/groovy/JaxRsFilterTest.groovy]: I do, and that is why it was reformatted when I updated other parts of the file.  It seems the original author doesn't have it configured.  Unless there's an easy way to validate it like we do with Java, I don't worry about it too much unless it starts thrashing a lot.

Comment (tylerbenson) [reply dd-java-agent/instrumentation/jax-rs-annotations-2/src/test/groovy/JaxRsFilterTest.groovy]: Looks like I lied... my settings were not the default.  I've fixed my settings, but not going to worry about reformatting now.... eventual consistency 😉

**Pull request merged**